core/src/env_platform/env_service.bal

```diff
@@ -349,15 +388,103 @@ service envservice on ep0 {
+        string authHeader = req.getHeader("Authorization");
+        [string, string]|error userInfoFromJWT = getUserInfoFromJWT(authHeader);
+        if (userInfoFromJWT is [string, string]) {
+            [string, string] [userId, userType] = userInfoFromJWT;
+            log:printDebug("User information - user ID: " + userId + ", user type: " + userType + ".");
+            if (body.sender.id == userId) {
+                boolean|error isApplicationRelatedToResult = isApplicationRelatedTo(userType, userId, applicationId);
+                if (isApplicationRelatedToResult is boolean && isApplicationRelatedToResult) {
+                    log:printDebug("Application is found with ID : " + applicationId + " related to the user with ID " 
+                        + userId + ".");
+                    boolean|error postCommentInApplicationResult = postCommentInApplication(applicationId,body);
+                    if (postCommentInApplicationResult is boolean && postCommentInApplicationResult) {
+                        log:printDebug("Succesfully posted the comment.");
+                        response.statusCode = http:STATUS_OK;
+                        response.setPayload({message: "Succesfully posted."}); 
+                    } else {
+                        response.statusCode = http:STATUS_INTERNAL_SERVER_ERROR;
+                        if (postCommentInApplicationResult is error) {
+                            log:printDebug("Error occured while posting the comment and the error occured is " 
+                            + postCommentInApplicationResult.toString() +".");
+                            response.setPayload(<@untainted>{message: postCommentInApplicationResult.reason()}); 
+                        } else {
+                            log:printDebug("Comment is not posted.");
+                            response.setPayload({message: "Comment is not posted."}); 
+                        }
+                    }
+                } else {
+                    if (isApplicationRelatedToResult is error) {
+                        log:printDebug("Error occured while checking the application is related to the user with ID " 
+                            + userId + ", and the error occured is " + isApplicationRelatedToResult.toString() + ".");
+                        response.statusCode = http:STATUS_INTERNAL_SERVER_ERROR;
+                        response.setPayload(<@untainted>{message: isApplicationRelatedToResult.reason()}); 
+                    } else {
+                        log:printDebug("Apploication is not found with ID : " + applicationId 
+                            + " related to the user with ID " + userId + ".");
+                        response.statusCode = http:STATUS_NOT_FOUND;
+                        response.setPayload({message: "Application is not found."}); 
+                    }
+                }
+            } else {
+                response.statusCode = http:STATUS_UNAUTHORIZED;
```

Comment:
I think we can merge the contents in this else block and the one after this into one since the error message is the same. @yashodgayashan can we use the logical _AND_ operator here?

Reply:
No it cannot be used here hence I have to add them separately.